hilti/toolchain/include/ast/operator.h

```diff
@@ -338,6 +338,9 @@ class Operator {
      */
     auto typename_() const { return _typename(); }
 
+    /** Returns a globally unique numeric identifier for the operator. */
+    uintptr_t identity() const { return reinterpret_cast<uintptr_t>(this); }
```

Comment:
One more question: why do we need this opaque value? Looks like it's just used as the index into `_resolved`. Could we just store `Operator*` in there directly?